src/app/shared/entry/descriptor-language.service.ts

```diff
@@ -60,28 +54,93 @@ export class DescriptorLanguageService {
     const combined$ = combineLatest([this.descriptorLanguages$, this.sessionQuery.entryType$]);
     this.filteredDescriptorLanguages$ = combined$.pipe(map((combined) => this.filterLanguages(combined[0], combined[1])));
   }
-  static toolDescriptorTypeEnumToDefaultDescriptorPath(descriptorType: ToolDescriptor.TypeEnum | null): string | null {
-    return DescriptorLanguageService.toolDescriptorTypeEnumToExtendedDescriptorLanguageBean(descriptorType).defaultDescriptorPath;
+
+  testParameterTypeEnumToToolDescriptorEnum(sourceFileType: SourceFile.TypeEnum | null): ToolDescriptor.TypeEnum | null {
+    return this.testSourceFileTypeEnumToExtendedDescriptorLanguageBean(sourceFileType).toolDescriptorEnum;
   }
 
-  static toolDescriptorTypeEnumToExtendedDescriptorLanguageBean(
-    descriptorType: ToolDescriptor.TypeEnum | null
-  ): ExtendedDescriptorLanguageBean {
+  testSourceFileTypeEnumToExtendedDescriptorLanguageBean(sourceFileType: SourceFile.TypeEnum | null): ExtendedDescriptorLanguageBean {
+    const foundExtendedDescriptorLanguageFromValue = extendedDescriptorLanguages.find(
+      (extendedDescriptorLanguage) => extendedDescriptorLanguage.testParameterFileType === sourceFileType
+    );
+    return foundExtendedDescriptorLanguageFromValue || extendedUnknownDescriptor;
+  }
+
+  sourceFileTypeEnumToExtendedDescriptorLanguageBean(sourceFileType: string | null): ExtendedDescriptorLanguageBean {
+    let foundExtendedDescriptorLanguageFromValue = extendedDescriptorLanguages.find((extendedDescriptorLanguage) =>
+      extendedDescriptorLanguage.descriptorFileTypes.find((f) => f === sourceFileType)
+    );
+    // Check if the source file type matches a test file type
+    if (!foundExtendedDescriptorLanguageFromValue) {
+      foundExtendedDescriptorLanguageFromValue = extendedDescriptorLanguages.find(
```

Comment:
This looks like the same code as the `testSourceFileTypeEnumToExtendedDescriptorLanguageBean` implemenation on line 63. If so, just call `testSourceFileTypeEnumToExtendedDescriptorLanguageBean`

Reply:
I see. It's confusing how we sometimes have enums and sometimes have strings (not your fault, I know it's already like this). We should try to clean that up some day. And overall these changes are a major improvement.

That said, it seems like the method name is misleading: `sourceFileTypeEnumTo...` -- the parameter is not an enum, it's a string, as you just pointed out. :)

Reply:
fixed the name